Doc/library/sqlite3.rst

```diff
@@ -1434,10 +1434,21 @@ case-insensitively by name:
 Using the connection as a context manager
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-Connection objects can be used as context managers
-that automatically commit or rollback transactions.  In the event of an
-exception, the transaction is rolled back; otherwise, the transaction is
-committed:
+Connection objects can be used as context managers that automatically commit or
+rollback open transactions when leaving the body of the context manager.
+If the body of the :keyword:`with` statement finishes without exceptions,
+the transaction is committed.
+If this commit fails, the transaction is rolled back.
+If the body of the ``with`` statement raise an exception which is not caught,
+the transaction is rolled back.
```

Comment:
Perhaps combine these, e.g. "If the commit fails, or if there was an uncaught exception in the body, the transaction is rolled back".